include/zephyr/kernel/thread.h

```diff
@@ -264,6 +264,9 @@ struct k_thread {
 
 	uint32_t   events;
 	uint32_t   event_options;
+
+	/** true if timeout must be prevented */
+	bool prevent_timeout;
```

Comment:
Can we name this something else?  I get that it "prevents handling of thread timeouts", but that relies on the reader understanding that a thread timeout is used exclusively within the kernel to wake up pended threads.  Maybe "no_wake", "no_timeout_unpend" or something similar?

Reply:
renamed prevent_timeout to no_wake_on_timeout now. Its a longer name but I think it justifies the potential confusion that you pointed out.